companion/flashing.md

```diff
@@ -41,43 +63,77 @@ Zadig will install the driver, and should report success. When done you can dism
-button and select a location for the file. This will automatically  
-build and download the latest available version of OpenTX for Taranis.  
-Close the Preferences window, and select the Burn -&gt; Write Flash  
-memory command. Make sure your radio is powered off and connected.  
-Browse for the file you just downloaded. It will be recognised  
-automatically and the version will be displayed.  If you wish to  
-replace the radio's splash screen, you can either load one from a file  
-or select it from the available library, and click the "Substitute  
-image in firmware" box.  Click "Burn to TX".  Once the progress bar is  
-done and the success prompt is shown you can dismiss it and disconnect  
-your radio :\) If you get an error about the dfu-util executable not  
-being found, open the flashing tool settings with the Burn -&gt;  
-Configure menu item. Browse for the dfu-util executable, which should  
-be: On Windows: In the OpenTX companion installation folder \(by  
-default C:\Program Files\companion9x\dfu-util.exe on 32-bit systems,  
-and C:\Program Files \(x86\)\companion9x\dfu-util.exe on 64-bit  
-systems\).  On Mac OS: `/Applications/OpenTX Companion  
-2.2.app/Contents/Resources/dfu-util`
+## Downloading The OpenTX Firmware
+
+The only recommended way to download the OpenTX firmware is using Companion. This process compiles/builds the firmware on the OpenTX servers and downloads a copy to the computer.
+
+To download the firmware follow these steps:
+
+* Radio Profile: Make sure the correct transmitter profile is selected
+* Build Options: Make sure the build options selected are the ones you want
+* Download Firmware: File -> Download
+* Click on Download firmware (The download could take a few minutes)
+* Save the firmware .bin file on the computer
+* You may be prompted to write the firmware to the transmitter
+* If you want to choose yes connect the transmitter to the computer in bootloader mode
+* Click on Yes
+* Select Check Hardware compatibility to make sure the firmware being written to the transmitter is for that particular hardware.
+* Click on Write to TX
+* Eject the mounted drives
+* Disconnect the USB cable
+* Power the transmitter off
+
+# Writing The OpenTX Firmware
+
+Connect the transmitter to the computer in bootloader mode.
+
+There are two options to write the firmware:
+* Using The Bootloader Menu
+* Using Companion
+
+## Using The Bootloader Menu
+
+* Place a copy of the firmware on the sd card in the FIRMWARE folder
+* Rename the firmware file so that the filename including the .bin extension is 35 characters or less.
+* Eject the two mounted drives
+* Disconnect the USB cable
+* Choose Write Firmware from the bootloader menu
+* Select the firmware to be written
+* Press ENT
+* Hold ENT down to start writing the firmware
+* When the “Writing Complete” message is displayed press EXIT
+* Power the transmitter off
+
+## Using Companion (dfu-util)
```

Comment:
This section is wrong.

In order for Companion to use dfu-util/zadig the radio has to be in STM Bootloader. So the radio has to be OFF and then connected to computer.

What you describe below is also possible, using OpenTX Bootloader like you described above. But in this case **dfu-util/zadig is not used**. This method uses USB mass storage mode (that is implemented by the OpenTX Bootloader) so no special driver needed for this.

Reply:
I think the confusion is that we have writing firmware under the driver section... the writing firmware may be clearer if put in its own section... When I wrote the *writing firmware instructions* I wrote if for a majority audience of FrSKY Tx purchased today and going forward... well purchased within the last year... for this audience informaing them about dfu/zadig etc. is unnecessary and adds confusion... many users forst want to know how to do something and only want to know the why and underlying details later on out of curiosity or have problems... This section makes more sense when arranged as it is in this article
http://rcdiy.ca/opentx-guide/introduction-to-opentx-companion/
Where I state 
Transmitter Read & Write

Connecting The Transmitter In Bootloader Mode

Most new transmitter such as the Taranis series do not require any additional software or drivers to be installed on the computer. One exception is the FrSKY Horus X12S which is covered here.

So how do I remove this section from here while retaining the other edits that have passed?

Reply:
I've browsed trough your page (link above) and there instructions are much more clear. I agree that skipping whole dfu-util/zadig is a good thing. 

But on the other hand  dfu-util/zadig has to be documented somewhere because it is needed for X12 and can be helpul/needed for X9D.

>So how do I remove this section from here while retaining the other edits that have passed?

Just move the dfu-util/zadig to a separate section.